integration-tests/smoke/auto_ocr_test.go

```diff
@@ -0,0 +1,181 @@
+
+		By("Deploy Registry and Registrar")
+		registry, registrar = actions.DeployAutoOCRRegistryAndRegistrar(
+			registryVersion,
+			registryConfig,
+			numberOfUpkeeps,
+			linkToken,
+			contractDeployer,
+			chainClient,
+		)
+
+		By("Create OCR Automation Jobs")
+		actions.CreateOCRKeeperJobs(chainlinkNodes, registry.Address(), network.ChainID)
+		nodesWithoutBootstrap := chainlinkNodes[1:]
+		ocrConfig := actions.BuildOCRConfigVars(nodesWithoutBootstrap, registryConfig, registrar.Address())
+		err = registry.SetConfig(defaultRegistryConfig, ocrConfig)
+		Expect(err).ShouldNot(HaveOccurred(), "Registry config should be be set successfully")
+
+		By("Deploy Consumers")
+		switch consumerContract {
+		case BasicCounter:
+			consumers, upkeepIDs = actions.DeployConsumers(
+				registry,
+				registrar,
+				linkToken,
+				contractDeployer,
+				chainClient,
+				numberOfUpkeeps,
+				linkFundsForEachUpkeep,
+				defaultUpkeepGasLimit,
+			)
+		}
+
+		if testToRun == BasicSmokeTest {
+			By("watches all the registered upkeeps perform and then cancels them from the registry")
+			Eventually(func(g Gomega) {
+				// Check if the upkeeps are performing multiple times by analysing their counters and checking they are greater than 10
+				for i := 0; i < len(upkeepIDs); i++ {
+					counter, err := consumers[i].Counter(context.Background())
+					g.Expect(err).ShouldNot(HaveOccurred(), "Failed to retrieve consumer counter for upkeep at index "+strconv.Itoa(i))
+					g.Expect(counter.Int64()).Should(BeNumerically(">=", int64(5)),
+						"Expected consumer counter to be greater than 0, but got %d", counter.Int64())
+					log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Number of upkeeps performed")
+				}
+			}, "5m", "1s").Should(Succeed())
+
+			// Cancel all the registered upkeeps via the registry
+			for i := 0; i < len(upkeepIDs); i++ {
+				err := registry.CancelUpkeep(upkeepIDs[i])
+				Expect(err).ShouldNot(HaveOccurred(), "Could not cancel upkeep at index "+strconv.Itoa(i))
+			}
+
+			err := chainClient.WaitForEvents()
+			Expect(err).ShouldNot(HaveOccurred(), "Error encountered when waiting for upkeeps to be cancelled")
+
+			var countersAfterCancellation = make([]*big.Int, len(upkeepIDs))
+
+			for i := 0; i < len(upkeepIDs); i++ {
+				// Obtain the amount of times the upkeep has been executed so far
+				countersAfterCancellation[i], err = consumers[i].Counter(context.Background())
+				Expect(err).ShouldNot(HaveOccurred(), "Failed to retrieve consumer counter for upkeep at index "+strconv.Itoa(i))
+				log.Info().Msg("Cancelled upkeep at index " + strconv.Itoa(i) + " which performed " +
+					strconv.Itoa(int(countersAfterCancellation[i].Int64())) + " times")
+			}
+
+			Consistently(func(g Gomega) {
+				for i := 0; i < len(upkeepIDs); i++ {
+					// Expect the counter to remain constant because the upkeep was cancelled, so it shouldn't increase anymore
```

Comment:
I think this was flaky behaviour in previous test too, but upkeep can be performed after it was cancelled if it was already in flight.

In v2.0, it can only happen at max once, so maybe assert that latestCounter <= countersAfterCancellation + 1?

Reply:
changed